src/migrer/core.clj

```diff
@@ -122,20 +200,91 @@
              (when-let [old-hash (get repeatable-hashes filename)]
                (= old-hash (md5sum sql)))))))
 
+(comment
+  (->> "-- :id \"aaa\"\n-- :pre [\"drop foobar;\" \"drop bax;\"]\ncreate table foo ();"
+      (clojure.string/split-lines)
+      (map (partial re-matches #"-- :([a-zA-Z]+) (.+)"))
+      (remove empty?)
+      (map rest)
+      (map (fn [[k v]] [(keyword k) (clojure.edn/read-string v)]))
+      (group-by first)
+      (map (fn [[k [[_ v]]]] [k v]))
+      (into {})
+      )
+
+  (re-matches #"-- :.+" "-- :id \"asdfa\"")
+  )
+
+(defn- extract-meta
```

Comment:
Expect the comments at the top of the file to be a comments with EDN values inside, and use regex to strip the comment characters, followed by parsing the data with clojure.edn/read-string. That should get rid of all the magic in this implementation.